core/corehttp/lazyseek.go

```diff
@@ -0,0 +1,60 @@
+package corehttp
+
+import (
+	"fmt"
+	"io"
+)
+
+// The HTTP server uses seek to determine the file size. Actually _seeking_ can
+// be slow so we wrap the seeker in a _lazy_ seeker.
+type lazySeeker struct {
+	reader io.ReadSeeker
+
+	size       int64
+	offset     int64
+	realOffset int64
+}
+
+func (s *lazySeeker) Seek(offset int64, whence int) (int64, error) {
+	switch whence {
+	case io.SeekEnd:
+		return s.Seek(s.size+offset, io.SeekStart)
```

Comment:
isn't it `s.size-offset` ?

Reply:
If we're seeking from the end, the offset will already be negative.

Reply:
Actually, this _was_ wrong. It should have been `s.size-1-offset`.